### PR TITLE
feat(jit): support introspection in JIT

### PR DESCRIPTION
**Pull request by laststylebender14**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/jit/request.rs [hunk 1]
```diff
@@ -19,15 +18,18 @@ pub struct Request<V> {
...
+            operation_name: value.operation_name.clone(),
+            variables: Variables::from_iter(
+                value
+                    .variables
+                    .iter()
+                    .map(|(k, v)| (k.to_string(), v.to_owned())),
+            ),
+            extensions: value.extensions.clone(),
         }
     }
```
Comment (tusharmath) [inline src/core/jit/request.rs:34]: Move the cloning logic outside of this function. Force the caller to clone if required.

IMPORTANT: Add a `NOTE: This is hot code and should allocate minimal memory`

File: src/core/jit/model.rs [hunk 2]
```diff
@@ -350,6 +350,7 @@ pub struct OperationPlan<Input> {
     nested: Vec<Field<Nested<Input>, Input>>,
     // TODO: drop index from here. Embed all the necessary information in each field of the plan.
     pub index: Arc<Index>,
+    is_introspection_query: bool,
```
Comment (tusharmath) [inline src/core/jit/model.rs:353]:
```suggestion
    pub is_introspection_query: bool,
```

File: src/core/jit/model.rs [hunk 3]
```diff
@@ -415,6 +424,10 @@ impl<Input> OperationPlan<Input> {
         self.operation_type == OperationType::Query
     }
 
+    pub fn is_introspection_query(&self) -> bool {
+        self.is_introspection_query
+    }
```
Comment (tusharmath) [inline src/core/jit/model.rs:429]: drop this, keep the field public.

Comment (laststylebender14) [reply src/core/jit/model.rs:429]: done

File: src/core/jit/model.rs (hunk 2 above)
Comment (laststylebender14) [reply src/core/jit/model.rs:353]: done

File: src/core/jit/request.rs (hunk 1 above)
Comment (laststylebender14) [reply src/core/jit/request.rs:34]: done

File: src/core/jit/graphql_executor.rs [hunk 4]
```diff
@@ -21,15 +23,42 @@ impl JITExecutor {
...
+        request.operation_name = self.operation_name.clone();
+        request.into()
+    }
+}
+
 impl Executor for JITExecutor {
     fn execute(&self, request: async_graphql::Request) -> impl Future<Output = Response> + Send {
-        let request = jit::Request::from(request);
+        let request = request.lift();
+        let jit_request = jit::Request::from(request.clone().take());
```
Comment (tusharmath) [inline src/core/jit/graphql_executor.rs:39]: Cloning should only be done if it's a introspection query.

Comment (laststylebender14) [reply src/core/jit/graphql_executor.rs]: done

**Review by tusharmath — APPROVED** [review]

**Pull request merged**